### PR TITLE
dapp filter by payee

### LINKED ISSUE #189: Dapps must be able to filter Providers by Payee
The Provider field `Payee` determines who gets paid when a captcha is answered - the Provider or the Dapp. Dapps must be able to select Providers based on this `Payee`. They should be able to choose Providers with payee of `Dapp`, `Provider` or `Any`

### PR DESCRIPTION
**Pull request by forgetso**
Fixes #189
- Store provider accounts by GovernanceStatus and Payee
- Only register providers with Payee::Dapp for now
- Only register dapps with DappPayee::Dapp for now
- add test
- fmt

### CODE REVIEW

File: contracts/lib.rs [hunk 1]
```diff
@@ -258,6 +276,8 @@ pub mod prosopo {
         contract: AccountId,
         owner: AccountId,
         value: Balance,
+        payee: DappPayee,
```
Comment (goastler) [inline contracts/lib.rs:279]: what are these for?

**Review by goastler — CHANGES_REQUESTED** [review]

File: contracts/lib.rs [hunk 2]
```diff
@@ -258,6 +276,8 @@ pub mod prosopo {
...
+        status: GovernanceStatus,
```
Comment (goastler) [inline contracts/lib.rs:280]: what is this for?

**Review by goastler — CHANGES_REQUESTED** [review]

File: contracts/lib.rs (hunk 2 above)
Comment (forgetso) [reply contracts/lib.rs:280]: Just extra information for the `DappRegister` event.

File: contracts/lib.rs (hunk 1 above)
Comment (forgetso) [reply contracts/lib.rs:279]: This is tested in `test_get_random_active_provider_dapp_any`

**Review by goastler — APPROVED** [review]

**Pull request merged**